src/coderunner/CodeRunner.ts

```diff
@@ -175,32 +179,76 @@ class PythonCodeRunner implements ICodeRunner {
         tests,
         additionalFiles,
         additionalFilesLoaded,
-        bookNode
+        bookNode,
+        sessionFiles
       );
     });
   };
 
   public drawTurtleExample = (
     additionalFilesLoaded: AdditionalFilesContents,
-    bookNode: BookNodeModel
+    bookNode: BookNodeModel,
+    sessionFiles: SessionFile[] = []
   ) => {
     if (this.turtleExamplePromiseResRej) {
       this.turtleExamplePromiseResRej.rej("Turtle drawing cancelled");
     }
 
     return new Promise<string>((res, rej) => {
       this.turtleExamplePromiseResRej = { res, rej };
-      this.runTurtleExample(additionalFilesLoaded, bookNode);
+      this.runTurtleExample(additionalFilesLoaded, bookNode, sessionFiles);
     });
   };
 
+  private additionalCodeForFiles = (
```

Comment:
extracted logic for creating the additional starter code (we had this in a few places)